src/main/java/de/tum/in/www1/artemis/service/ExamService.java

```diff
@@ -369,4 +383,13 @@ private Exercise selectRandomExercise(SecureRandom random, ExerciseGroup exercis
 
         return generatedParticipations;
     }
+
+    private Exam breakSerializationCycleExercises(Exam exam) {
```

Comment:
I think it would be better if you do the following instead and delete this method

In `ExerciseGroup` add the annotation to the `exercises` association
`@JsonIgnoreProperties("exerciseGroup")`

This is also how it works for the association between `Course` and `Exercise`